src/EventListener/InitializeSystemListener.php

```diff
@@ -275,29 +274,32 @@ class_alias('Contao\\ModuleLoader', 'ModuleLoader');
      */
     private function setRelativePath($basePath)
     {
-        Environment::set('path', $basePath);
-
-        define('TL_PATH', Environment::get('path')); // backwards compatibility
+        define('TL_PATH', $basePath); // backwards compatibility
     }
 
     /**
      * Sets the default language.
+     *
+     * @param Request $request
      */
-    private function setDefaultLanguage()
+    private function setDefaultLanguage(Request $request = null)
     {
-        if (!isset($_SESSION['TL_LANGUAGE'])) {
-            $langs = Environment::get('httpAcceptLanguage');
+        if (!$this->session->has('TL_LANGUAGE')) {
```

Comment:
By `root` @aschempp meant the root of `$_SESSION`. Because the SF2 Session handling stores everything in `$_SESSION['_sf2_attributes']` by default. But again, it will work because we also set it on the session service later :)